src/Microsoft.AspNetCore.Mvc.Core/Internal/AssemblyPartDiscoveryProvider.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    internal static class AssemblyPartDiscoveryProvider
+    {
+        public static void ConfigureApplicationParts(ApplicationPartManager partManager, string entryPointAssemblyName)
```

Comment:
This seems like it should be a static factory method on `ApplicationPartManager`

Reply:
We can't make it a factory since the earliest we can call this, we may have an instance of `ApplicationPartManager`.